tests/test_command_handler.py

```diff
@@ -96,6 +96,26 @@ async def test_configure(self) -> None:
         )
         self.assert_response(common.ResponseCode.OK)
         assert self.configuration == self.command_handler.configuration
+        # Send the same configuration again which should lead to an error.
+        await self.command_handler.handle_command(
+            command=common.Command.CONFIGURE, configuration=self.configuration
+        )
+        self.assert_response(common.ResponseCode.ALREADY_STARTED)
+        # Make sure to stop so a new config command can be sent.
+        await self.command_handler.stop_sending_telemetry()
+        # Give time to the telemetry_task to get cancelled.
+        await asyncio.sleep(0.5)
```

Comment:
Is it possible to make `stop_sending_telemetry()` wait until it has truly stopped before returning? That would eliminate the need for this sleep.

If not, is there some task you can await in the command handler or in the mock server, to know that the connection has been dropped? That would be more robust than waiting for a fixed amount of time.

If there is no existing natural task for this, please consider adding such a Task, Future, or asyncio.Event. If the use is only for unit tests, that's OK, just document it as such. I sometimes add a task for this that a unit test can set to a new Future and that the library code sets done if not already done when the thing happens.

Reply:
It turns out that the calls to `sleep` weren't necessary so I removed them all.